lib/SILGen/SILGenBridging.cpp

```diff
@@ -1363,8 +1363,10 @@ emitObjCThunkArguments(SILGenFunction &SGF, SILLocation loc, SILDeclRef thunk,
   }
 
   // We don't know what to do with indirect results from the Objective-C side.
-  assert(objcFnTy->getNumIndirectFormalResults() == 0
-         && "Objective-C methods cannot have indirect results");
+  assert((SGF.F.getRepresentation() ==
+              SILFunctionType::Representation::CFunctionPointer ||
+          objcFnTy->getNumIndirectFormalResults() == 0) &&
+         "Objective-C methods cannot have indirect results");
```

Comment:
I tried compiling the following code, where `S` is a non-trivial C++ type, but the compiler rejects it:

```
@objc
class C0 : NSObject {
  // error: method cannot be marked @objc because its result type cannot be represented in Objective-C
  // note: non-trivial C++ classes cannot be represented in Objective-C
  @objc
  func m1() -> S {
    return S()
  }
}
```

The diagnostic was added in https://github.com/apple/swift/pull/69343.

Is there another way to trigger the assertion?

Reply:
That seems quite wrong.  I mean, we can have this restriction if we need to subset things, but it is absolutely not the case that non-trivial C++ classes cannot be represented in Objective-C.

Reply:
At any rate, please write the code assuming that this is required to work, even if there's currently a restriction about it.

Reply:
Yes, I removed the assertion altogether.